Liber Usualis - mei3/liberupdatev5.py

```diff
@@ -0,0 +1,240 @@
+    epi_dict = {"horizontal": "h", "vertical": "v"}
+
+    # Flags for tracking certain conditions
+    theres_an_episema = False
+    theres_a_pb = False
+    theres_a_sb = False
+    nc_flag = False
+    neume_flag = False
+
+    # Extra attributes for neumes and notes
+    extra_nc = {}
+    extra_neume = {}
+
+    # Set MEI version to 5.0
+    root.attrib["meiversion"] = "5.0"
+
+    # Iterate over all elements in the tree
+    for child in root.iter("*"):
+        child.tag = ET.QName(child).localname
+
+        # Handle graphic elements (e.g., change href to target)
+        if child.tag.endswith("graphic"): #changes some stuff in the headers
+            for att in list(child.attrib):
+                if att.endswith("href"):                    
+                    child.attrib('target', child.attrib.pop(att))
+        
+        # Handle page breaks
+        elif child.tag.endswith("pb"): #adds in pagebreak info from layout element (which gets deleted later)
+            child.tag = "peeb"
+            for att in child.attrib:                
+                if att.endswith("pageref"):                    
+                    pageref = child.attrib.pop(att)
+                    break
+            for child_2 in root.iter():
+              for attr, value in child_2.attrib.items():               
+                if attr.endswith("id") and value == pageref:
+                        child.set("n", child_2.attrib["n"])
+                        break                 
+            pa_bre = ET.tostring(child)
+            theres_a_pb = True
+            child.tag = "TODELETE"
+        
+        # Handle system breaks
+        elif child.tag.endswith("sb"):
+            child.tag = "seeb"
+            for att in child.attrib:
+                if att.endswith("systemref"):                    
+                    systemref = child.attrib.pop(att)
+                    break
+            for child_2 in root.iter():               
+              for attr, value in child_2.attrib.items():
+                if attr.endswith("id") and value == systemref:
+                        child.set("facs", child_2.attrib["facs"])
+                        break
+            sy_bre = ET.tostring(child)           
+            theres_a_sb = True
+            child.tag = "TODELETE"
+
+        # Moves system breaks and page breaks inside layer
+        elif child.tag.endswith("layer"):
+            if theres_a_pb:
+                theres_a_pb = False
+                child.append(ET.fromstring(pa_bre))
+            if theres_a_sb:
+                theres_a_sb = False
+                child.append(ET.fromstring(sy_bre))
+
+        # Handle zone elements (adjusts negative ulx and lrx attributes)
+        elif child.tag.endswith("zone"):
+            label = ""
+            for attr, value in child.attrib.items():
+                if attr.endswith("ulx") and int(value) < 0:
+                    label = f"{label} ulx = {value} "
```

Comment:
Your `label` attribute will now always have a leading and trailing space if not empty...this doesn't seem right.

Reply:
Label has been removed